Update reverse-proxy.md

punkyard opened the pull request:
few suggestions to help reading ^^

docjyJ commented on reverse-proxy.md:
````diff
@@ -145,15 +145,15 @@ https://<your-nc-domain>:443 {
 }
 ```
-The Caddyfile is a text file called `Caddyfile` (no extension) which – if you should be running Caddy inside a container – should usually be created in the same location as your `compose.yaml` file prior to starting the container.
+The Caddyfile is a text file called `Caddyfile` (no extension) which – if you should be running Caddy inside a container – should usually be created in the same location as your `docker-compose.yaml` file prior to starting the container.
````
Since compose files are used by [multiple backends](https://github.com/compose-spec/compose-spec#implementations), `compose.yaml` is more relevant. See:
https://github.com/compose-spec/compose-spec/blob/231b09c30d339e950c0da17fe5bdc793366b8fde/03-compose-file.md?plain=1#L11-L13

szaimen replied:
```suggestion
The Caddyfile is a text file called `Caddyfile` (no extension) which – if you should be running Caddy inside a container – should usually be created in the same location as your `compose.yaml` file prior to starting the container.
```

docjyJ commented on reverse-proxy.md:
```diff
@@ -393,19 +393,19 @@ server {
 <summary>click here to expand</summary>
 
-First, please make sure that the environmental variables `PUID` and `PGID` in the compose.yaml file for NPM are either unset or set to `0`.
+First, make sure the environmental variables `PUID` and `PGID` in the docker-compose.yaml file for NPM are either unset or set to `0`.
```
Same

szaimen replied:
```suggestion
First, make sure the environmental variables `PUID` and `PGID` in the `compose.yaml` file for NPM are either unset or set to `0`.
```

docjyJ commented on reverse-proxy.md:
````diff
@@ -393,19 +393,19 @@ server {
 ```
 
-⚠️ **Please note:** Look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
+⚠️ **Please note:** look in [here](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
````
I find it weird, but I'm not a native English speaker so I'm not sure.

szaimen replied:
Also not sure but lets leave it like this for now

docjyJ commented on reverse-proxy.md:
````diff
@@ -528,35 +526,35 @@ httpServer.on('upgrade', (req, socket, head) => {
 ```
 
-⚠️ **Please note:** Look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
+⚠️ **Please note:** look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
````
Here it has not been changed?

docjyJ commented on reverse-proxy.md:
```diff
@@ -717,11 +714,11 @@ Config examples for other reverse proxies are currently not documented. Pull req
 After adjusting your reverse proxy config, use the following command to start AIO:<br>
 
-(For a docker-compose example, see the example further [below](#inspiration-for-a-docker-compose-file).)
+(For a docker-compose.yml example, see the example further [below](#inspiration-for-a-docker-compose-file).)
```
```suggestion
(For a `compose.yaml` example, see the example further [below](#inspiration-for-a-docker-compose-file).)
```

punkyard commented on reverse-proxy.md:
```diff
@@ -2,23 +2,23 @@
 **Note:** The maintainers of AIO noticed that this documentation could be improved to make it easier to follow. All contributions that improve this are very welcome!
 
-A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a web server that enables computers on the internet to access a service in a [private subnet](https://en.wikipedia.org/wiki/Private_network).
+A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically the entry point of a web server. It allows other computers to access the [private subnet](https://en.wikipedia.org/wiki/Private_network) of the web server over the Internet.
```
sorry, my mistake, somehow my sentence was truncated:

"A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a software service that can be thought of as the gateway of a web server that enables computers on the internet to access a service or web-site in a [private subnet] of that web server.

docjyJ commented on reverse-proxy.md:
```diff
@@ -2,23 +2,23 @@
+A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a software service that can be thought of as the gateway of a web server. It enables computers on the internet to access a service or web-site in a [private subnet](https://en.wikipedia.org/wiki/Private_network) of that web server.
 
-**Please note:** Publishing the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! Instead, the main goal is to publish Nextcloud with a valid certificate to the public internet which is **not** running inside the mastercontainer but in a different container! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get- ...
+**Please note:** Deploying the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).
```
I think publishing is better than deploying because it is deploying the interface without publishing it.

szaimen replied:
```suggestion
**Please note:** Publishing the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).
```

szaimen commented on reverse-proxy.md:
```diff
@@ -2,23 +2,23 @@
+**Please note:** Deploying the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).
 
-In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to specify the port that AIO's Apache container shall use, add a specific config to your web server or reverse proxy and modify the startup command a bit. All examples below will use port `11000` as example `APACHE_PORT` which will be exposed on the host to receive unen ...
+In order to run Nextcloud behind a web server / reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to 1. specify the port that AIO's integrated Apache container shall use 2. add a specific config to your web server / reverse proxy 3. modify the startup command a bit. All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private se ...
```
you could also use the <br> tag

szaimen replied:
```suggestion
In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to <br>1. specify the port that AIO's integrated Apache container shall use <br>2. add a specific config to your web server or reverse proxy <br>3. modify the startup command a bit. <br>All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private network to receive unencrypted HTTP traffic from the reverse proxy. **Advice:** If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.
```

szaimen commented:
Hi @punkyard would you mind addressing the review by @docjyJ or shall we take over? :)

punkyard commented:
> Hi @punkyard would you mind addressing the review by @docjyJ or shall we take over? :)

Hi Simon, I guess you guys should take over. It was for me a first time to contribute to the project with an outlook as a user. As I think that sometimes formulation can be difficult to conceptualise for beginners, like myself. 
I do hope I managed to bring something useful here. 🙏

szaimen commented:
> > Hi @punkyard would you mind addressing the review by @docjyJ or shall we take over? :)
> 
> Hi Simon, I guess you guys should take over. It was for me a first time to contribute to the project with an outlook as a user. As I think that sometimes formulation can be difficult to conceptualise for beginners, like myself. I do hope I managed to bring something useful here. 🙏

Thanks a lot @punkyard 💙 These kind of contributions are exactly what we are looking for :)

szaimen reviewed (approved):
LGTM. @docjyJ also for you? :)

docjyJ commented on reverse-proxy.md:
```diff
@@ -1,22 +1,23 @@
+> The maintainers of AIO noticed that this documentation could be improved to make it easier to follow. All contributions that improve this are very welcome!
 
-A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a web server that enables computers on the internet to access a service in a [private subnet](https://en.wikipedia.org/wiki/Private_network).
+A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a software service that can be thought of as the gateway of a web server. It enables computers on the internet to access a service or web-site in a [private subnet](https://en.wikipedia.org/wiki/Private_network) of that web server.
```
```suggestion
A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is a software service that acts as a gateway between services and a client. It is commonly used to allow a client connected to the Internet to access a website located in the [private subnet](https://en.wikipedia.org/wiki/Private_network) of that web server.
```

docjyJ commented on reverse-proxy.md:
```diff
@@ -1,22 +1,23 @@
+**Please note:** Publishing the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).
 
-In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to specify the port that AIO's Apache container shall use, add a specific config to your web server or reverse proxy and modify the startup command a bit. All examples below will use port `11000` as example `APACHE_PORT` which will be exposed on the host to receive unen ...
+In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to <br>1. specify the port that AIO's integrated Apache container shall use <br>2. add a specific config to your web server or reverse proxy <br>3. modify the startup command a bit. <br>All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed ...
```
```suggestion
In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to:
1. specify the port that AIO's integrated Apache container shall use
2. add a specific config to your web server or reverse proxy
3. modify the startup command a bit.
All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private network to receive unencrypted HTTP traffic from the reverse proxy.
> [!IMPORTANT] 
> If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.
```
> simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.

No, "simply" use port 443.... Nextcloud is exposed in HTTPS...

docjyJ commented on reverse-proxy.md:
```diff
@@ -201,7 +202,7 @@ Although it does not seem like it is the case but from AIO perspective a Cloudfl
 
 1. Install the Cloudflare Tunnel on the same machine where AIO will be running on and point the Tunnel with the domain that you want to use for AIO to `http://localhost:11000`.<br>
 ⚠️ **Please note:** Look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
```
```suggestion
⚠️ **Please note:** look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
```
consistent

docjyJ reviewed (changes requested):
otherwise LGTM

docjyJ commented:
It's ready.

Otherwise do you know of any small free VPS?
I'd like to put an AIO on the dev branch to test.

docjyJ commented:
I don't know why I can create an account on Oracle...

szaimen commented:
> I don't know why I can create an account on Oracle...

you cannot?

docjyJ commented:
https://www.youtube.com/watch?v=SsK4YIIR69w

exatly my issue

szaimen commented:
Thanks for all the reviews @docjyJ 💙
